pkg/controller/deployment/sync.go

```diff
@@ -632,8 +592,12 @@ func calculateStatus(allRSs []*extensions.ReplicaSet, newRS *extensions.ReplicaS
 
 // isScalingEvent checks whether the provided deployment has been updated with a scaling event
 // by looking at the desired-replicas annotation in the active replica sets of the deployment.
-func (dc *DeploymentController) isScalingEvent(d *extensions.Deployment) (bool, error) {
-	newRS, oldRSs, err := dc.getAllReplicaSetsAndSyncRevision(d, false)
+//
+// rsList should come from getReplicaSetsForDeployment(d).
+// podMap should come from getPodMapForReplicaSets(rsList).
+// These are passed around to avoid repeating expensive API calls.
```

Comment:
The only mutating action we were doing before is the relabeling of replica sets and pods but that shouldn't happen only in rare cases (adopting old replica sets maybe).